draft/0006-pat.md

```diff
@@ -0,0 +1,125 @@
+---
+Start date: 2024-08-15
+Author(s): wangsijie
+Revision: 1
+---
+
+# Personal Access Token (PAT)
+
+## Overview
+
+This RFC proposes an extension to Logto's authentication system to support Personal Access Tokens (PATs). This feature aims to provide users with a secure and convenient way to authenticate automated processes and integrations. PATs are long-lived tokens that can be used by non-interactive clients to exchange for short-lived access tokens through token exchange endpoints first introduced in RFC-0005 impersonation.
+
+## Motivation
+
+The current authentication flow in Logto is optimized for interactive user sessions. However, there are scenarios where users need to authenticate non-interactive processes or third-party integrations. These include:
+
+1. Automated scripts or bots that perform actions on behalf of the user.
+2. Integration with external tools and services that require API access.
+3. Long-running processes that outlast typical session durations.
+
+## Personal Access Token (PAT) Creation
+
+The client makes a post request to the tenant's Management API `/api/users/:userId/personal-access-tokens` to create a PAT for the user. The following parameters are included in the HTTP request entity-body using the `application/json` format.
+
+1. `name`: REQUIRED. A user-defined name for the PAT to identify its purpose, should be unique in the user's account.
+2. `scopes`: OPTIONAL. The requested scopes for the PAT, limited to the user's permissions. If not provided, the PAT will have all the user's permissions.
+
+If the request to create a APT is successful, the tenant's Management API returns the PAT in the response.
+
+1. `name`: REQUIRED. The name of the PAT.
+2. `token`: REQUIRED. The value of the PAT.
+3. `scopes`: OPTIONAL. The scopes of the PAT.
+
+### Example PAT creation
+
+```http
+POST /api/users/123/personal-access-tokens HTTP/1.1
+Host: tenant.logto.app
+Content-Type: application/json
+Authorization: Bearer <access-token-with-management-api-access>
+
+{
+  "name": "My PAT",
+  "scopes": ["profile", "read:logs", "write:logs"]
```

Comment:
Should we throw error when the scopes of a PAT is a super set of the user's scopes?

Reply:
Per discussion, scope support is removed.